### PR TITLE
zsh: refine module

### PR DESCRIPTION
**Pull request by Infinisil**
This fixes a problem I've been getting with the changes in #63 and refines the module some more. The problem was that since ~/.zshenv set ZDOTDIR but $ZDOTDIR/.zshenv didn't, running zsh inside a zsh would fail to set the env vars, since zsh looks at $ZDOTDIR/.zshenv if ZDOTDIR is set.

Another change/fix is that now if zsh is disabled, no zsh config is being built, whereas previously plugins and oh-my-zsh were still being built.

### CODE REVIEW

**Comment by Infinisil** [comment]
I have tested this with bash and zsh as envVarSetter, and both with dotDir null and non-null.

File: modules/programs/zsh.nix [hunk 1]
```diff
@@ -11,6 +11,19 @@ let
...
+
+  toEnvVarsStr = vars: concatStringsSep "\n" (
+    mapAttrsToList export vars
+  );
+
+  envVarsStr = toEnvVarsStr config.home.sessionVariables;
+
+  aliasesStr = concatStringsSep "\n" (
+    mapAttrsToList (k: v: "alias ${k}='${v}'") cfg.shellAliases
+    );
```
Comment (uvNikita) [inline modules/programs/zsh.nix]: Looks like this bracket has an extra indentation.

File: modules/programs/zsh.nix [hunk 2]
```diff
@@ -272,6 +265,26 @@ in
...
+      home.sessionVariables.ZDOTDIR = zdotdir;
+    })
+    (mkIf (config.home.sessionVariableSetter == "zsh" && cfg.dotDir == null) {
+      home.file.".zshenv".text = ''
+        ${envVarsStr}
+      '';
+    })
+    (mkIf (config.home.sessionVariableSetter == "zsh" && cfg.dotDir != null) {
+      # When dotDir is set, only use ~/.zshenv to export ZDOTDIR, $ZDOTDIR/.zshenv for the rest
+      # This is so that if ZDOTDIR happens to be already set correctly (by e.g. spawning a zsh inside a zsh), all env vars still get exported
```
Comment (uvNikita) [inline modules/programs/zsh.nix]: Could you please wrap this comment to 80 characters as well? This line looks very long.

**Comment by Infinisil** [comment]
Did the formatting adjustments

**Comment by uvNikita** [comment]
Thank you! Rebased into master: https://github.com/rycee/home-manager/commit/db55e596d2010025882f3f6067a886bff0ddbae8.

**Pull request closed without merging**